deps/src/main/java/com/microsoft/azure/sdk/iot/deps/twin/TwinCollection.java

```diff
@@ -360,17 +360,12 @@ private static void addMetadata(TwinCollection twinCollection, Map<? extends Str
             } else if (key.equals(TwinMetadata.LAST_UPDATED_BY_DIGEST)) {
                 lastUpdatedByDigest = (String) entry.getValue();
             } else {
-                Object valueInCollection = twinCollection.get(key);
-                if (valueInCollection == null) {
-                    // If the property (key) exists in metadata but not in twinCollection metadata is inconsistent.
-                    throw new IllegalArgumentException("Twin metadata is inconsistent for property: " + key);
-                }
-
                 TwinMetadata twinMetadata = TwinMetadata.tryExtractFromMap(entry.getValue());
                 if (twinMetadata != null) {
```

Comment:
Maybe I don't understand this scenario very well, but if it was null, wouldn't that mean we want to delete the key from the twin? Should we have code that handles when it is null? Or, is it null in this case because we couldn't parse it. And if that is true, should we log something that says we couldn't parse something so we're doing nothing with it.